### PR TITLE
feat: Added Authentication

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
This is a PR for the following changes:
- Added Login, Signup, and Logout functionality
- Added Form Validation for user data on sign-up
- Added context for Authentication Data (like Auth tokens), and User Data (including liked videos, watch later, watch history etc)
- Added Helper hooks like useSignUpHandler and useLoginHandler for Auth.
- Also added useful wrapper functions for common API calls to fetch data.

Preview link for the changes: https://deploy-preview-2--techtv.netlify.app/

### CODE REVIEW

**Review by samyakshah3008 — APPROVED** [review]
Clean code and UI 
Good for approval

File: src/common/helpers/AccountService.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+const getUserData = async (authToken) => {
+    try {
+        const response = await axios.get("/api/user/account", { headers: { authorization: authToken }});
+        if(response.status === 200) {
+            return response.data;
+        }
+    } catch(err) {
+        console.log("getUserData: Error in fetching user account data", err.response.data.errors[0]);
```
Comment (manish591) [inline src/common/helpers/AccountService.js:10]: use console.error

**Review by manish591 — APPROVED** [review]
The code is looking good. I like that you have added comments in your code which makes code more readable.

**Review by Aman-Jain1996 — APPROVED** [review]
Code is working as expected, good to merge

**Pull request merged**